tls/s2n_kem.h

```diff
@@ -78,19 +88,8 @@ extern const struct s2n_kem s2n_kyber_512_r3;
 extern const struct s2n_kem s2n_kyber_768_r3;
 extern const struct s2n_kem s2n_kyber_1024_r3;
 
-/* x25519 based tls13_kem_groups require EVP_APIS_SUPPORTED */
-/* Kyber758+ requires S2N_LIBCRYPTO_SUPPORTS_KYBER */
-#if defined(S2N_LIBCRYPTO_SUPPORTS_KYBER) && EVP_APIS_SUPPORTED
-    #define S2N_SUPPORTED_KEM_GROUPS_COUNT 6
-#elif defined(S2N_LIBCRYPTO_SUPPORTS_KYBER) && !EVP_APIS_SUPPORTED
-    #define S2N_SUPPORTED_KEM_GROUPS_COUNT 4
-#elif !defined(S2N_LIBCRYPTO_SUPPORTS_KYBER) && EVP_APIS_SUPPORTED
-    #define S2N_SUPPORTED_KEM_GROUPS_COUNT 2
-#else
-    #define S2N_SUPPORTED_KEM_GROUPS_COUNT 1
-#endif
-
-extern const struct s2n_kem_group *ALL_SUPPORTED_KEM_GROUPS[S2N_SUPPORTED_KEM_GROUPS_COUNT];
+#define S2N_KEM_GROUPS_COUNT 6
+extern const struct s2n_kem_group *ALL_SUPPORTED_KEM_GROUPS[S2N_KEM_GROUPS_COUNT];
```

Comment:
will do.

Reply:
hm, turns out we can't because we use `S2N_KEM_GROUPS_COUNT` to tell the compiler the size of `ALL_SUPPORTED_KEM_GROUPS`. i also tried eliding the pre-declared size (the compiler should be smart enough to figure this out by multiplying the number of elements by the size of a ptr, right?) but then i got complaints about variable sizing elsewhere where we use `S2N_KEM_GROUPS_COUNT` to define the size of other arrays.

this shouldn't be too onerous to update if we ever touch this array again; the compiler will catch it. we do this for [named ECs](https://github.com/aws/s2n-tls/blob/main/crypto/s2n_ecc_evp.h#L53), presumably for the same reason.

Reply:
Darn, you're right :/ But at least the compiler catches it.